Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -10,17 +10,17 @@
 using BoogiePL = Microsoft.Boogie;
 using System.Runtime.Caching;
 using System.Diagnostics;
-using System.Net.Mime;
 using System.Runtime.CompilerServices;
-using Core;
-using Microsoft.Boogie.Houdini;
 using VCGeneration;
 
 namespace Microsoft.Boogie
 {
-  public class ExecutionEngine : IDisposable {
+  public record ProcessedProgram(Program Program, Action<VCGen, Implementation, VerificationResult> PostProcessResult) {
```

Comment:
How many different instances of this record do you create?  I'm more used to seeing abstract methods than fields of type `Action`, so if there are not too many different post-processors an abstract class may be better?